pkg/api/customization/authn/user_store.go

```diff
@@ -111,21 +115,36 @@ func (s *userStore) Create(apiContext *types.APIContext, schema *types.Schema, d
 		return nil, err
 	}
 
-	if id, ok := created[types.ResourceFieldID].(string); ok {
-		var principalIDs []interface{}
-		if pids, ok := created[client.UserFieldPrincipalIDs].([]interface{}); ok {
-			principalIDs = pids
-		}
-		created[client.UserFieldPrincipalIDs] = append(principalIDs, "local://"+id)
-		created, err = s.Update(apiContext, schema, created, id)
-		if err != nil {
-			return created, err
+	for x := 0; x < 3; x++ {
+		if id, ok := created[types.ResourceFieldID].(string); ok {
+			var principalIDs []interface{}
+			if pids, ok := created[client.UserFieldPrincipalIDs].([]interface{}); ok {
+				principalIDs = pids
+			}
+			created[client.UserFieldPrincipalIDs] = append(principalIDs, "local://"+id)
+			created, err = s.Update(apiContext, schema, created, id)
+			if err != nil {
+				if httperror.IsConflict(err) {
+					time.Sleep(3 * time.Second)
+
+					created, err = s.ByID(apiContext, schema, id)
+					if err != nil {
+						return nil, fmt.Errorf("error retrieving by id: %v", err)
+					}
+
+					continue
+				} else {
+					return nil, err
+				}
+			}
+
+			delete(created, client.UserFieldPassword)
+
+			return created, nil
 		}
 	}
 
-	delete(created, client.UserFieldPassword)
-
-	return created, err
+	return nil, fmt.Errorf("could not update user within maximum retries")
```

Comment:
YIKES! This resolved by deleting the 
```
delete(created, client.UserFieldPassword)
```
I added yesterday, which must remain

Reply:
the fact that we are retrying is an implementation detail that I would  like to hide from the user. In fact, now that I think about it, I think that if all updates fail, we need to *delete* the user. and then return a 500 error saying failed to create user

Reply:
I moved that to line 141.  I believe that that keeps the functionality the same (because we're now returning nil on 147).

Reply:
discussed offline. this thread of comments no longer valid. we are now going to retry and if that fails, just log an error and return the user anyway. we are also going to add logic to the create function of the user lifecycle to ensure the local principal id is set.

Reply:
sleep will also be reduced to 100 ms * loop index